File: medium/2270. Number of Ways to Split Array/main.ts
const waysToSplitArray = (nums: number[]): number => {
  let result = 0, prefixSum = 0

  let arraySum = nums.reduce((sum, num) => sum + num, 0);

  for (let i = 0; i < nums.length - 1; i++) {
    prefixSum += nums[i];
    if (prefixSum >= arraySum - prefixSum) {
      result++;
    }
  }

  return result
};

File: medium/2270. Number of Ways to Split Array/main.go
package main

func waysToSplitArray(nums []int) int {
	var result, prefixSum, arraySum int

	for _, num := range nums {
		arraySum += num
	}

	for i := 0; i < len(nums)-1; i++ {
		prefixSum += nums[i]

		if prefixSum >= arraySum-prefixSum {
			result++
		}
	}

	return result
}
